minard/views.py

```diff
@@ -108,22 +110,37 @@ def state(run=None):
     if run_state['tubii'] is not None:
         tubii_state = detector_state.get_tubii_state(run_state['tubii'])
 
-    crates_state =[None]*20
+    crates_state = [None]*20
     for iCrate in range(20):
         if run_state['crate'+str(iCrate)] is not None:
             crates_state[iCrate] = detector_state.get_crate_state(run_state['crate'+str(iCrate)])
 
     if not any(crates_state):
-        crates_state = None;
-
-    return render_template('state.html',run=run,
-                                        run_state = run_state,
-                                        detector_control_state = detector_control_state,
-                                        mtc_state = mtc_state,
-                                        caen_state = caen_state,
-                                        tubii_state = tubii_state,
-                                        crates_state = crates_state,
-                                        err = None)
+        crates_state = None
+    trigger_scan = None
+    if run_state['timestamp'] is not None:
+        trigger_scan = detector_state.get_trigger_scan_for_run(run)
+
+    oncolor = request.args.get('oncolor')
+    offcolor = request.args.get('offcolor')
+    colors = [oncolor, offcolor]
```

Comment:
Do you need these colors in the flask rendering or just on the javascript side? If just on the javascript side, a cleaner way to do this would be to get the url parameters in javascript. I already have a function which loads them into a global variable called `url_params` (see https://github.com/snoplus/minard/blob/master/minard/static/js/minard.js). To use them it's as easy as `url_params.oncolor` (see https://github.com/snoplus/minard/blob/master/minard/static/js/snostream.js#L2).